client/src/contexts/user/UserContext.tsx

```diff
@@ -0,0 +1,31 @@
+import { createContext, PropsWithChildren, useState } from 'react';
+
+type User = {
+  id: number;
+  firstname: string;
+  lastname: string;
+  email: string;
+  role: {
+    id: number;
+    label: string;
+  };
+};
+
+type UserContextType = {
+  user: User | null;
+  setUser: React.Dispatch<React.SetStateAction<User | null>>;
+};
```

Comment:
We don't want to put those two types in a separate file?

Reply:
yes, and user can be taken from Codegen